tests/Unit/CMSoftwareUpdatePoint.tests.ps1

```diff
@@ -0,0 +1,532 @@
+        -DSCModuleName $script:dscModuleName `
+        -DSCResourceName $script:dscResourceName `
+        -ResourceType 'Mof' `
+        -TestType 'Unit'
+
+    # Import Stub function
+    $script:moduleRoot = Split-Path -Parent (Split-Path -Parent $PSScriptRoot)
+    Import-Module (Join-Path -Path $PSScriptRoot -ChildPath 'Stubs\ConfigMgrCBDscStub.psm1') -Force -WarningAction SilentlyContinue
+}
+
+function Invoke-TestCleanup
+{
+    Restore-TestEnvironment -TestEnvironment $script:testEnvironment
+}
+
+Invoke-TestSetup
+
+#Begin Testing
+try
+{
+    InModuleScope $script:dscResourceName {
+        $moduleResourceName = 'ConfigMgrCBDsc - DSC_CMSoftwareUpdatePoint'
+
+        Describe 'ConfigMgrCBDsc - DSC_CMSoftwareUpdatePoint\Get-TargetResource' -Tag 'Get'{
+            BeforeAll{
+                $getInput = @{
+                    SiteCode       = 'Lab'
+                    SiteServerName = 'CA01.contoso.com'
+                }
+
+                $getSUPReturn = @{
+                    SiteCode = 'Lab'
+                    Props    = @(
+                        @{
+                            PropertyName = 'AllowProxyTraffic'
+                            Value        = 0
+                        }
+                        @{
+                            PropertyName = 'IsINF'
+                            Value        = '0'
+                        }
+                        @{
+                            PropertyName = 'IsIntranet'
+                            Value        = '1'
+                        }
+                        @{
+                            PropertyName = 'SSLWSUS'
+                            Value        = 0
+                        }
+                        @{
+                            PropertyName = 'UseProxy'
+                            Value        = 0
+                        }
+                        @{
+                            PropertyName = 'UseProxyForADR'
+                            Value        = 0
+                        }
+                        @{
+                            PropertyName = 'WSUSAccessAccount'
+                            Value2       = $null
+                        }
+                        @{
+                            PropertyName = 'WSUSIISPort'
+                            Value        = '8530'
+                        }
+                        @{
+                            PropertyName = 'WSUSIISSSLPort'
+                            Value        = '8531'
+                        }
+                    )
+                }
+
+                Mock -CommandName Import-ConfigMgrPowerShellModule
+                Mock -CommandName Set-Location
+            }
+
+            Context 'When retrieving software update point settings' {
+
+                It 'Should return desired result when software update point is not currently installed' {
+                    Mock -CommandName Get-CMSoftwareUpdatePoint -MockWith { $null }
```

Comment:
this wouldn't need the -MockWith { $null } there is no carry over to this IT block

Reply:
removed -MockWith {$null}